### ISSUE TITLE
More robust edit

### ISSUE THREAD

**Issue by Jip-Hop** [Jip-Hop/jailmaker#91]
Improve `jlmkr edit`

1. copy config to tmp file
2. edit the tmp file
3. after save, parse and validate new config
4. if invalid, don't apply
5. else move original config to config.bak (may be valid syntax, but could still cause jail not to start)
6. move tmp file over original config
7. check if contents have changed after edit
8. only show hint to restart the jail if content has changed

Alternatively to moving files around and creating .bak files, perhaps creating a snapshot of the jails dir (or jail subdir in case #80 and #82 are done) would be even better.

**Comment by Jip-Hop** [comment]
Wontfix. A user can configure regular snapshots from the TrueNAS GUI to protect themselves in case they mess up the config.

**Comment by Lockszmith-GH** [comment]
ooh, incorporating zfs snapshots into the edit process - now that's a novel 'safety first' idea.
We can use zfs properties to determine whether jailmaker has that on or off - so no new persistence introduced.
If you're not averse to the idea, I might try to tackle it in the future. (I'm also looking at preserving the config files in an archive dir, when removing a jail - I always find myself wanting to review that configuration I modified - AFTER I've deleted the jail)
